code/string/sub-vs-chomp.rb

```diff
@@ -0,0 +1,17 @@
+require 'benchmark/ips'
+
+SLUG = 'YourSubclassType'
+
+def slow
+  SLUG.sub(/Type$/, '')
```

Comment:
`$` matches the end of a line. Regular expressions in Ruby default to multiline mode. The regex should be changed to `/Type\z/`. 

Some examples:

```
irb [2.1.5] (sshaw)$ s = "abc\nxyz\n"
=> "abc\nxyz\n"
irb [2.1.5] (sshaw)$ s.sub /abc$/, ""
=> "\nxyz\n"
irb [2.1.5] (sshaw)$ s.sub /abc\z/, ""
=> "abc\nxyz\n"
irb [2.1.5] (sshaw)$ s.sub /abc\Z/, ""
=> "abc\nxyz\n"
irb [2.1.5] (sshaw)$ s.sub /xyz\Z/, ""
=> "abc\n\n"
irb [2.1.5] (sshaw)$ s.sub /xyz\z/, ""
=> "abc\nxyz\n"
```